src/rng.rs

```diff
@@ -316,25 +320,25 @@ pub trait Fill {
     fn try_fill<R: Rng + ?Sized>(&mut self, rng: &mut R) -> Result<(), Error>;
 }
 
-macro_rules! impl_fill_each {
+macro_rules! impl_fill_each_uninit {
     () => {};
     ($t:ty) => {
-        impl Fill for [$t] {
+        impl Fill for [mem::MaybeUninit<$t>] {
             fn try_fill<R: Rng + ?Sized>(&mut self, rng: &mut R) -> Result<(), Error> {
                 for elt in self.iter_mut() {
-                    *elt = rng.gen();
+                    *elt = mem::MaybeUninit::new(rng.gen());
                 }
                 Ok(())
             }
         }
```

Comment:
Implementing `Fill for [MaybeUninit<bool>]` (etc.) is okay in itself, but I don't think it very useful. In order to *use* it with uninitialized memory, `assume_init` is needed somewhere, which thus requires the user use `unsafe` code (and based on the assumption that `fill` succeeded). I mean, users *can* use this API correctly, but not easily, and it isn't difficult to implement equivalent functionality in userspace anyway.